### PR TITLE
test: use unit tests for getUserMedia constraints converter

### PR DESCRIPTION
**Pull request by fippo**
uses sinon.stub to make assertions about how the constraints
converter works in different browsers and versions.

@jan-ivar PTAL -- this was one of the more scary tests, I find this a bit easier to reason about.
IIRC this test was one of the reasons we are still using webkitGetUserMedia.

### CODE REVIEW

File: test/unit/getusermedia-constraints.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        }
+      };
+      window.navigator.getUserMedia(legacy);
+      expect(window.navigator.mozGetUserMedia).to.have.been.calledWith(legacy);
+    });
+  });
+
+  describe('in Firefox 38+', () => {
+    beforeEach(() => {
+      utils.browserDetails.version = 53;
```
Comment (jan-ivar) [inline test/unit/getusermedia-constraints.js]: The old version had 38 here.

**Review by jan-ivar — APPROVED** [review]
Nice! sinon seems a good fit for this sort of thing.

File: test/unit/getusermedia-constraints.js (hunk 1 above)
Comment (fippo) [reply test/unit/getusermedia-constraints.js]: changed

**Comment by fippo** [comment]
sinon is a lot less hassle than doing this yourself for sure ;-)

**Pull request merged**